File: 34.py
def smallest_rotation(s: str) -> str:
    n = len(s)
    best = s
    for i in range(1, n):
        rotated = s[i:] + s[:i]
        if rotated < best:
            best = rotated
    return best

# Driver code
if __name__ == "__main__":
    s = input().strip()
    print(smallest_rotation(s))

File: 34.js
function smallestRotation(s) {
    let best = s;
    for (let i = 1; i < s.length; i++) {
        let rotated = s.slice(i) + s.slice(0, i);
        if (rotated < best) best = rotated;
    }
    return best;
}

// Driver code
const fs = require("fs");
const s = fs.readFileSync(0, "utf8").trim();
console.log(smallestRotation(s));
